packages/core/src/scp-cf/destination-accessor.ts

```diff
@@ -153,348 +168,420 @@ export async function getDestinationFromDestinationService(
+    );
+  }
+
+  public static async getDestinationFromDestinationService(
+    name: string,
+    options: DestinationOptions
+  ): Promise<Destination | null> {
+    const decodedUserJwt = await DestinationAccessor.getDecodedUserJwt(options);
+    const providerToken = await DestinationAccessor.getProviderToken(options);
+    const da = new DestinationAccessor(
       name,
-      isolation
+      options,
+      decodedUserJwt,
+      providerToken
     );
-    if (subscriberDestinationCache) {
+    const destinationResult =
+      (await da.searchSubscriberAccountForDestination()) ??
+      (await da.searchProviderAccountForDestination());
+
+    if (destinationResult?.fromCache) {
+      return destinationResult.destination;
+    }
+
+    if (destinationResult) {
       logger.info(
-        'Successfully retrieved destination from destination service cache for subscriber destinations.'
+        'Successfully retrieved destination from destination service.'
       );
-      return subscriberDestinationCache;
+    } else {
+      logger.info('Could not retrieve destination from destination service.');
+      return null;
     }
-  }
 
-  const { userJwt, ...optionsWithoutJwt } = options;
-  const providerToken = await serviceToken('destination', optionsWithoutJwt);
-  const decodedProviderJwt = await decodeJwt(providerToken);
+    let { destination } = destinationResult;
+    if (destination.authentication === 'OAuth2SAMLBearerAssertion') {
+      destination = await da.addOAuthSamlAuth(destination);
+    }
+    if (destination.authentication === 'ClientCertificateAuthentication') {
+      destination = await da.addClientCertAuth();
+    }
 
-  if (options.useCache) {
-    providerDestinationCache = destinationCache.retrieveDestinationFromCache(
-      decodedProviderJwt,
-      name,
-      isolation
+    const withProxySetting = await da.addProxyConfiguration(destination);
+    await da.updateDestinationCache(
+      withProxySetting,
+      destinationResult.account
     );
-    if (
-      providerDestinationCache &&
-      (selectionStrategy === alwaysProvider ||
-        (decodedUserJwt &&
-          isIdenticalTenant(decodedUserJwt!, decodedProviderJwt)))
-    ) {
+    return withProxySetting;
+  }
+
+  private static searchServiceBindingForDestination(
+    name: string
+  ): Destination | undefined {
+    logger.info('Attempting to retrieve destination from service binding.');
+    try {
+      const destination = destinationForServiceBinding(name);
+      logger.info('Successfully retrieved destination from service binding.');
+      return destination;
+    } catch (error) {
+      logger.info(error.message);
+      logger.info('Could not retrieve destination from service binding.');
       logger.info(
-        'Successfully retrieved destination from destination service cache for provider destinations.'
+        'If you are not using SAP Extension Factory, this information probably does not concern you.'
+      );
```

Comment:
I would rather recommend adding one error message. It seems unnecessary to separate this and could be misinterpreted as you see the error first.
```ts
 logger.info(`Could not retrieve destination from service binding. If you are not using SAP Extension Factory, this information probably does not concern you. Original error: ${error.message}`);
```

Reply:
yes